### PR TITLE
#166187843 Upgrade project to Django 2.1

### PR DESCRIPTION
**Pull request by neelxie**
#### What does this PR do?

Upgrade Project from Django 1.10 to 2.1

#### Description of Task to be completed?

- Upgrade libraries in the requirements.txt file.
- Fix all errors arising from the library upgrades.
- Run the authentication migrations for the app authors.

#### How should this be manually tested?

- Do a `git fetch origin ft-upgrade-django-166187843`
- Then a git checkout into the `ft-upgrade-django-166187843` branch
- then run the project with
`$ python3 manage.py runserver`

#### Any background context you want to provide?

- Some Django features like `django.conf.urls` have been deprecated and I have had to use `path` instead.
- I have added the pylint and autopep8 library to the project to deal with autoformatting and to have the project conform to the best quality standards.
- I have had to make the project migrations for the app to be able to run.

#### What are the relevant pivotal tracker stories?

[#166187843](https://www.pivotaltracker.com/story/show/166187843)

### CODE REVIEW

**Comment by HabibSentongo** [comment]
It's better under **How should this be manually tested?** you start with creating a directory as the first step before someone creating a virtual environment.

**Comment by alexxsanya** [comment]
@neelxie the link to clone should be https://github.com/andela/ah-backend-realers.git **not** https://github.com/andela/ah-backend-realers/tree/ft-upgrade-django-166187843 as you specified in PR notice.
After that, you also need to add a step of switching to the **ft-upgrade-django-166187843** branch.

**Comment by neelxie** [comment]
> @neelxie the link to clone should be https://github.com/andela/ah-backend-realers.git **not** https://github.com/andela/ah-backend-realers/tree/ft-upgrade-django-166187843 as you specified in PR notice.
> After that, you also need to add a step of switching to the **ft-upgrade-django-166187843** branch.

Thanks @alexxsanya  let me fix that.

**Review by alexxsanya — APPROVED** [review]
@neelxie I followed through PR notice, the project is upgraded to Django 2.1 as in the screenshot

Screenshot 2019-05-27 at 16 11 15

**Comment by cdvx** [comment]
> > @neelxie the link to clone should be https://github.com/andela/ah-backend-realers.git **not** https://github.com/andela/ah-backend-realers/tree/ft-upgrade-django-166187843 as you specified in PR notice.
> > After that, you also need to add a step of switching to the **ft-upgrade-django-166187843** branch.
> 
> Thanks @alexxsanya let me fix that.

@alexxsanya , @neelxie , in this case you want reviewers to just fetch the branch not clone the whole repo, `git fetch origin ft-upgrade-django-166187843` then `git checkout ft-upgrade-django-166187843` to test it out locally. I say this because I imagine all contributers have cloned and set up the repo locally and hence recloning it would be a repitition.

File: authors/apps/authentication/migrations/0001_initial.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='User',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('password', models.CharField(max_length=128, verbose_name='password')),
+                ('last_login', models.DateTimeField(blank=True, null=True, verbose_name='last login')),
+                ('is_superuser', models.BooleanField(default=False, help_text='Designates that this user has all permissions without explicitly assigning them.', verbose_name='superuser status')),
```
Comment (jennizalwango) [inline authors/apps/authentication/migrations/0001_initial.py:21]: Hey @neelxie compared to pep8 standards this looks so long, actually most of these lines here are abit long

**Comment by neelxie** [comment]
> > > @neelxie the link to clone should be https://github.com/andela/ah-backend-realers.git **not** https://github.com/andela/ah-backend-realers/tree/ft-upgrade-django-166187843 as you specified in PR notice.
> > > After that, you also need to add a step of switching to the **ft-upgrade-django-166187843** branch.
> > 
> > 
> > Thanks @alexxsanya let me fix that.
> 
> @alexxsanya , @neelxie , in this case you want reviewers to just fetch the branch not clone the whole repo, `git fetch origin ft-upgrade-django-166187843` then `git checkout ft-upgrade-django-166187843` to test it out locally. I say this because I imagine all contributers have cloned and set up the repo locally and hence recloning it would be a repitition.

@cdvx How would you suggest we handle this particular scenerio.

**Comment by alexxsanya** [comment]
> > > > @neelxie the link to clone should be https://github.com/andela/ah-backend-realers.git **not** https://github.com/andela/ah-backend-realers/tree/ft-upgrade-django-166187843 as you specified in PR notice.
> > > > After that, you also need to add a step of switching to the **ft-upgrade-django-166187843** branch.
> > > 
> > > 
> > > Thanks @alexxsanya let me fix that.
> > 
> > 
> > @alexxsanya , @neelxie , in this case you want reviewers to just fetch the branch not clone the whole repo, `git fetch origin ft-upgrade-django-166187843` then `git checkout ft-upgrade-django-166187843` to test it out locally. I say this because I imagine all contributers have cloned and set up the repo locally and hence recloning it would be a repitition.
> 
> @cdvx How would you suggest we handle this particular scenerio.

I think under this & similar scenarios, **git fetch**  is more appropriate. thanks, @cdvx for this highlight

**Comment by cdvx** [comment]
> > > > @neelxie the link to clone should be https://github.com/andela/ah-backend-realers.git **not** https://github.com/andela/ah-backend-realers/tree/ft-upgrade-django-166187843 as you specified in PR notice.
> > > > After that, you also need to add a step of switching to the **ft-upgrade-django-166187843** branch.
> > > 
> > > 
> > > Thanks @alexxsanya let me fix that.
> > 
> > 
> > @alexxsanya , @neelxie , in this case you want reviewers to just fetch the branch not clone the whole repo, `git fetch origin ft-upgrade-django-166187843` then `git checkout ft-upgrade-django-166187843` to test it out locally. I say this because I imagine all contributers have cloned and set up the repo locally and hence recloning it would be a repitition.
> 
> @cdvx How would you suggest we handle this particular scenerio.



Since these reviews are done by contributers, they already have the code base set up, so to test it out manually, they can just fetch the branch, check it out and test locally. You can also go ahead to include any necessary information that your work introduced that they may need. But that can also be added under the `Any background context you want to provide?` section.

**Comment by Emmanuelaaron** [comment]
Under ```How this PR can be tested```, you need to specify which branch the reviewer should switch to when testing your PR manually. Other than that you're good to go

**Comment by cdvx** [comment]
> Under `How this PR can be tested`, you need to specify which branch the reviewer should switch to when testing your PR manually. Other than that you're good to go

@Emmanuelaaron , please let us endeavor to review the code more than the PR description

File: authors/apps/authentication/migrations/0001_initial.py (hunk 1 above)
Comment (neelxie) [reply authors/apps/authentication/migrations/0001_initial.py:21]: @jennizalwango Kindly note that this file and it's contents are auto generated when I run the migrations command; so I have no control over its standards.

**Review by neelxie — COMMENTED** [review]
> Under `How this PR can be tested`, you need to specify which branch the reviewer should switch to when testing your PR manually. Other than that you're good to go

@Emmanuelaaron Thank you for pointing that out, but we have resolved this particular issue in the above comments.

**Review by jennizalwango — APPROVED** [review]

**Comment by neelxie** [comment]
> Any background context you want to provide?

@cdvx Thank you and I have added a little more information in that line under `Any background context you want to provide?`

**Comment by Emmanuelaaron** [comment]
> > Under `How this PR can be tested`, you need to specify which branch the reviewer should switch to when testing your PR manually. Other than that you're good to go
> 
> @Emmanuelaaron Thank you for pointing that out, but we have resolved this particular issue in the above comments.

okay noted.

**Review by Emmanuelaaron — APPROVED** [review]
Great work @neelxie,
I checked the code, tested it on my local machine and it's running well.
I think you're good to go

File: authors/settings.py [hunk 2]
```diff
@@ -137,7 +137,7 @@
     'EXCEPTION_HANDLER': 'authors.apps.core.exceptions.core_exception_handler',
     'NON_FIELD_ERRORS_KEY': 'error',
 
-    'DEFAULT_AUTHENTICATION_CLASSES': (
-        'authors.apps.authentication.backends.JWTAuthentication',
-    ),
+    # 'DEFAULT_AUTHENTICATION_CLASSES': (
```
Comment (cdvx) [inline authors/settings.py]: @neelxie , please shed more light to the team on why you commented this part out. Also, please ensure that your [PR naming](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices) and [commit message](https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices) conform to the team conventions. To give more context, your commit message should have a body.

Comment (neelxie) [reply authors/settings.py]: @cdvx sorry this was deprecated code I forgot to remove but let me re-push and make the changes.

**Review by HabibSentongo — APPROVED** [review]
With these final changes, I think everything is okay now.

**Pull request merged**